qt/shutdowndlg.py

```diff
@@ -0,0 +1,67 @@
+# SPDX-FileCopyrightText: © 2025 Huaide Jiang
+#
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# This file is part of the program "Back In Time" which is released under GNU
+# General Public License v2 (GPLv2). See LICENSES directory or go to
+# <https://spdx.org/licenses/GPL-2.0-or-later.html>.
+""" Module about showing a warning dialog before shutting down """
+import os
+from PyQt6.QtWidgets import (QApplication,
+                             QDialog,
+                             QLabel,
+                             QPushButton,
+                             QVBoxLayout)
+from PyQt6.QtCore import QTimer
+
+
+class ShutdownWarningDialog(QDialog):
+    """A UI class for a shutting down window."""
+    def __init__(self, countdown):
+        super().__init__()
+        self.countdown = countdown
+
+        # Initialize UI components
+        self.setWindowTitle("Back In Time - Shutdown Warning")
+        self.setFixedSize(300, 150)
+        self.label = QLabel(f"Shutdown in {self.countdown}s", self)
```

Comment:
Fill the lable in the update_countdown() method only.

```suggestion
        self.label = QLabel('', self)
```

Reply:
This one is needed, otherwise in the first second there'll be nothing in the window.

Reply:
I understand. But than just make sure the update-label message is called once from the __init__() method.

    self.label = QLabel('', self)
    self.update_countdown()

My goal here is to reduce redundancies in the code.